controllers/disruption_controller.go

```diff
@@ -192,53 +193,73 @@ func (r *DisruptionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 				r.recordEventOnDisruption(instance, chaosv1beta1.EventDisruptionStuckOnRemoval, "", "")
 			}
 
-			isCleaned, err := r.cleanDisruption(ctx, instance)
-			if err != nil {
-				return ctrl.Result{}, fmt.Errorf("error cleaning disruption: %w", err)
-			}
+			if instance.IsReadyToRemoveFinalizer(r.FinalizerDeletionDelay) {
+				// we reach this code when all the cleanup pods have succeeded and we waited for finalizerDeletionDelay
+				// we can remove the finalizer and let the resource being garbage collected
+				r.log.Infow("all chaos pods are cleaned up; removing disruption finalizer")
+				r.recordEventOnDisruption(instance, chaosv1beta1.EventDisruptionFinished, "", "")
 
-			// if not cleaned yet, requeue and reconcile again in 15s-20s
-			// the reason why we don't rely on the exponential backoff here is that it retries too fast at the beginning
-			if !isCleaned {
-				requeueAfter := time.Duration(randSource.Intn(5)+15) * time.Second //nolint:gosec
+				r.DisruptionsWatchersManager.RemoveAllWatchers(instance)
+				controllerutil.RemoveFinalizer(instance, chaostypes.DisruptionFinalizer)
 
-				r.log.Infow(fmt.Sprintf("disruption has not been fully cleaned yet, re-queuing in %v", requeueAfter))
+				if err := r.Client.Update(ctx, instance); err != nil {
+					return ctrl.Result{}, fmt.Errorf("error removing disruption finalizer: %w", err)
+				}
 
-				return ctrl.Result{
-					Requeue:      true,
-					RequeueAfter: requeueAfter,
-				}, r.Client.Update(ctx, instance)
-			}
+				// send reconciling duration metric
+				r.handleMetricSinkError(r.MetricsSink.MetricCleanupDuration(time.Since(instance.ObjectMeta.DeletionTimestamp.Time), []string{"disruptionName:" + instance.Name, "namespace:" + instance.Namespace}))
```

Comment:
Hmm, we're now adding 20s onto the cleanup duration metrics

Reply:
I was initially going for the fact that this is the time it took to actually delete the disruption from the signal to the end so I didn't modify it.

Reply:
If we don't like it, we can change it later